### PR TITLE
Refactored all likely exceptions to inherit from LarkError

### PR DESCRIPTION
**Pull request by erezsh**
Also improved some error messages.

Adds `ConfigurationError` to complement `GrammarError`.

Based on #773 by @ThatXliner

### CODE REVIEW

File: lark/lark.py [hunk 1]
```diff
@@ -277,24 +282,25 @@ def __init__(self, grammar, **options):
...
-            assert self.options.parser in disambig_parsers, (
-                'Only %s supports disambiguation right now') % ', '.join(disambig_parsers)
+            assert_config(self.options.parser, ('earley', 'cyk'), "%r doesn't support disambiguation. Use one of these parsers instead: %s")
 
         if self.options.priority == 'auto':
             self.options.priority = 'normal'
 
         if self.options.priority not in _VALID_PRIORITY_OPTIONS:
-            raise ValueError("invalid priority option: %r. Must be one of %r" % (self.options.priority, _VALID_PRIORITY_OPTIONS))
+            raise ConfigurationError("invalid priority option: %r. Must be one of %r" % (self.options.priority, _VALID_PRIORITY_OPTIONS))
```
Comment (MegaIng) [inline lark/lark.py:300]: This can be `assert_config`

File: lark/lark.py [hunk 2]
```diff
@@ -277,24 +282,25 @@ def __init__(self, grammar, **options):
...
         assert self.options.ambiguity not in ('resolve__antiscore_sum', ), 'resolve__antiscore_sum has been replaced with the option priority="invert"'
         if self.options.ambiguity not in _VALID_AMBIGUITY_OPTIONS:
-            raise ValueError("invalid ambiguity option: %r. Must be one of %r" % (self.options.ambiguity, _VALID_AMBIGUITY_OPTIONS))
```
Comment (MegaIng) [inline lark/lark.py:297]: This can also be `assert_config`

File: lark/exceptions.py [hunk 3]
```diff
@@ -7,9 +7,16 @@ class LarkError(Exception):
...
 
 
+class ConfigurationError(LarkError, ValueError):
+    pass
+
+
 class GrammarError(LarkError):
     pass
 
+class GrammarError_Value(LarkError):
```
Comment (MegaIng) [inline lark/exceptions.py:17]: Why does this exist?

**Review by MegaIng — CHANGES_REQUESTED** [review]

**Comment by MegaIng** [comment]
Also, stubs

File: lark/exceptions.py (hunk 3 above)
Comment (erezsh) [reply lark/exceptions.py:17]: Life is mysterious

**Comment by erezsh** [comment]
I don't think it's important that everything uses `assert_config`, it's just a convenience function.

**Review by MegaIng — APPROVED** [review]

**Pull request merged**